.github/workflows/build.yml

```diff
@@ -0,0 +1,85 @@
+# Copyright (c) 2022, Kry10 Limited
+# SPDX-License-Identifier: BSD-2-Clause
+
+name: Build
+
+on:
+  pull_request:
+    # Currently aspirational
+    branches:
+      - master
+  push:
+    branches:
+      - ci-riscv64
+  schedule:
+    - cron: '0 17 * * 6'
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout graph-refine
+        if: github.event_name != 'schedule'
+        uses: actions/checkout@v3
+      - name: Checkout graph-refine (schedule)
+        if: github.event_name == 'schedule'
+        uses: actions/checkout@v3
+        with:
+          ref: ci-riscv64
+      - name: Login to GitHub Container Registry
```

Comment:
I haven't checked the usage limits/implications of the GitHub registry. For other CI jobs we're currently using the dockerhub registry directly on the seL4 org there. I can make the org secret for that available fro this repo, but happy to use the GH registry for now for testing and defer this bit to later. If the GH registry turns out to be better, it might even make sense to move all seL4 CI containers there. In any case we should make a note to consolidate at some point.

Reply:
AFAICT, GH registry has no limits for public images. I put these on the GH registry since I already had the access, and the images are primarily intended for use by GH CI. As you say, it would be easy to move them. Fetching images from the GH registry to GH CI does seem to be pretty fast. Not that it matters for workflows that take hours to run...

Reply:
Cool, thank. Maybe I should indeed move all the other CI images to GH as well. It would make sense that these are faster, being on the same infrastructure etc (and docker-pull does take quite long for some of the shorter checks). In any case, good for now.